engine/src/main/java/org/pentaho/di/www/ExecuteJobServlet.java

```diff
@@ -202,26 +204,29 @@ public void doGet( HttpServletRequest request, HttpServletResponse response ) th
     Repository repository;
     try {
       repository = openRepository( repOption, userOption, passOption );
-    } catch ( KettleException ke ) {
+    } catch ( KettleRepositoryNotFoundException krnfe ) {
+      // Repository not found.
+      response.setStatus( HttpServletResponse.SC_NOT_FOUND );
       String message = BaseMessages.getString( PKG, "ExecuteJobServlet.Error.UnableToFindRepository", repOption );
-
-      if ( ke.getCause() instanceof ClientTransportException ) {
-        // Authentication Error.
-        response.setStatus( HttpServletResponse.SC_UNAUTHORIZED );
-        message = BaseMessages.getString( PKG, "ExecuteJobServlet.Error.Authentication", getContextPath() );
-        out.println( new WebResult( WebResult.STRING_ERROR, message ) );
-      } else if ( ke.getMessage().contains( message ) ) {
-        // Repository not found.
-        response.setStatus( HttpServletResponse.SC_NOT_FOUND );
-        out.println( new WebResult( WebResult.STRING_ERROR, message ) );
-      } else {
-        // Something unexpected occurred.
-        response.setStatus( HttpServletResponse.SC_INTERNAL_SERVER_ERROR );
-        message = BaseMessages.getString(
-          PKG, "ExecuteJobServlet.Error.UnexpectedError", Const.CR + Const.getStackTracker( ke ) );
-        out.println( new WebResult( WebResult.STRING_ERROR, message ) );
+      out.println( new WebResult( WebResult.STRING_ERROR, message ) );
+      return;
+    } catch ( KettleException ke ) {
+      // Authentication Error.
+      if (  ke.getCause() instanceof ExecutionException ) {

```

Comment:
Are there tests for these cases?

Reply:
There is a test to validate response error message, when a KettleAuthenticationException is thrown while connecting into a server.
https://github.com/pentaho/pentaho-kettle/blob/146e2160759081d40b451a35a11d953188a72e6f/engine/src/test/java/org/pentaho/di/www/ExecuteJobServletTest.java#L206